### PR TITLE
Feature / Form

### PR DESCRIPTION
**Pull request by Henrts**
Using `react-hook-forms` to build a generic Form component.

### CODE REVIEW

File: src/modules/Chip/Chip.story.tsx [hunk 1]
```diff
@@ -28,3 +30,25 @@ export const ChipIcon: React.FC<{
     />
   );
 };
+
+const DerivedChip = CloseableChip.extendStyles(() => ({
```
Comment (diogo-mendes) [inline src/modules/Chip/Chip.story.tsx]: Why so many changes to the chip? Is your branch rebased with master? Maybe that's the reason why you are apparently re-introducing old code

File: src/modules/TextInput/ErrorTextInput.tsx [hunk 2]
```diff
@@ -36,9 +36,9 @@ export const ErrorTextInput: React.FC<IProps & WithStylesProps> = ({
         {...props}
         hasError={error !== null && error !== undefined}
       />
-      {error && typeof error === "string" && (
-        <div className={cx(styles.errorLabel)}>{error}</div>
-      )}
+      <div className={cx(styles.errorLabel)}>
```
Comment (diogo-mendes) [inline src/modules/TextInput/ErrorTextInput.tsx]: Is this change intended? Do we want to have the error label element even without an error?

File: src/modules/TextInput/TextInput.tsx [hunk 3]
```diff
@@ -136,39 +137,35 @@ export const TextInput: React.FC<PropsWithChildren<
       >
         {label}
       </legend>
-      {label && (
-        <span ref={hiddenLabel} className={cx(styles.hiddenLabel)}>
+      <span ref={hiddenLabel} className={cx(styles.hiddenLabel)}>
```
Comment (diogo-mendes) [inline src/modules/TextInput/TextInput.tsx]: Same question as above: if the label is not necessarily gonna be passed, why render this element?

File: src/modules/Form/Form.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  form: {
+    display: "flex",
+    flexDirection: "column"
+  }
+});
+
+export interface IFormProps extends StandardProps<"form"> {
+  children: ReactElement[];
+  onSubmitFunc: (values: { [key: string]: string | boolean }) => void;
+  toSubmit?: {
```
Comment (diogo-mendes) [inline src/modules/Form/Form.tsx]: What does this prop do?

File: src/modules/TextInput/ErrorTextInput.tsx (hunk 2 above)
Comment (Henrts) [reply src/modules/TextInput/ErrorTextInput.tsx]: @nunoliveira12

Comment (nunoliveira12) [reply src/modules/TextInput/ErrorTextInput.tsx]: To keep the space of the error so it doesnt jump. There are alternatives to this

File: src/modules/TextInput/TextInput.tsx (hunk 3 above)
Comment (nunoliveira12) [reply src/modules/TextInput/TextInput.tsx]: label is always passed as "" if not specified. I just preferred not to keep conditions on render

File: src/modules/Checkbox/Checkbox.tsx [hunk 5]
```diff
@@ -143,8 +143,12 @@ const Checkbox: React.FC<IProps & WithStylesProps> = ({
     [onChange, checked, disabled]
   );
 
-  const handleClick = useCallback(() => {
+  const handleClick = useCallback((e?: React.MouseEvent) => {
     const input = inputRef.current;
+    if (e) {
+      e.stopPropagation();
```
[the remark this answers is not in the document]
Comment (diogo-mendes) [reply src/modules/Checkbox/Checkbox.tsx]: Like we discussed irl, I think only `e.preventDefault()` is necessary so we can remove `e.stopPropagation()`

File: src/modules/TextInput/TextInput.tsx (hunk 3 above)
Comment (diogo-mendes) [reply src/modules/TextInput/TextInput.tsx]: Yeah I understand that, that way you are forcing the element to always be rendered. But why? Is it for a similar reason you mentioned above, i.e. having consistent spacing? Contrary to the error label though, this label will not be potentially changing right?
This is a minor thing anyway 🙂
